src/stirling/testing/common.h

```diff
@@ -180,6 +186,41 @@ class Timeout {
   std::chrono::time_point<std::chrono::steady_clock> start_;
 };
 
+template <typename TRecord>
+bool RecordsContains(const std::vector<TRecord>& records, const std::vector<TRecord>& expected) {
+  for (const auto& expected_record : expected) {
+    bool in_records = false;
+    for (const auto& r : records) {
+      if (expected_record == r) {
+        in_records = true;
+        break;
+      }
+    }
+    if (!in_records) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// WaitAndExpectRecords calls `get_records` repeatedly until all of the `expected` records are
+// contained in the response to `get_records`. `TRecord` is required to have a operator== overload,
+// and suggested to have a PrintTo overload, see overloads.h for examples. `TGetRecords`
+// should be a function that returns a std::vector<TRecord>.
+template <typename TRecord, typename TGetRecords>
+std::vector<TRecord> WaitAndExpectRecords(
+    TGetRecords get_records, std::vector<TRecord> expected,
+    std::chrono::nanoseconds sleep_time = std::chrono::milliseconds{200}) {
+  std::vector<TRecord> records;
+  Timeout t;
+  while (!RecordsContains(records, expected) && !t.TimedOut()) {
+    records = get_records();
+    std::this_thread::sleep_for(sleep_time);
+  }
+  EXPECT_THAT(records, IsSupersetOf(expected));
```

Comment:
The new assertions are no longer strictly matching, correct? Was there flakiness with strict matching or did it just better match with this new style of testing (looping until all the records exist)?

Reply:
There are some tests that weren't doing strict matches and some that were, I felt like standardizing on this made sense, but open to opinions

Reply:
Sounds good, just wanted to understand the rationale.

I agree with opting for the less strict matches when it's not crucial to the specific test scenario. When we deflaked a bunch of trace bpf tests back in Jan/Feb, moving to the less strict matching helped in a few scenarios.

Reply:
FWIW the size checks are still there for the ones that had them